src/register/scause.rs

```diff
@@ -3,82 +3,14 @@
 use bit_field::BitField;
 use core::mem::size_of;
 
+pub use crate::register::mcause::{Interrupt, Exception, Trap};
```

Comment:
`Interrupt` and `Exception` should be different for `mcause` and `scause`

Reply:
I see the reserved fields pose a problem in the future, reverted